model/adb_auth.go

```diff
@@ -44,3 +72,191 @@ FROM adb_users
 
 	return *adbUser, nil
 }
+
+func GetUsersJSON(db *sqlx.DB, options GetUserOptions) ([]UserJSON, error) {
+	if options.ID != 0 {
+		return nil, errors.New("GetUsersJSON: Cannot include ID in options")
+	}
+
+	return getUsersJSON(db, options)
+}
+
+func getUsersJSON(db *sqlx.DB, options GetUserOptions) ([]UserJSON, error) {
+	users, err := GetUsers(db, options)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return buildUserJSONArray(users), nil
+}
+
+func GetUsers(db *sqlx.DB, options GetUserOptions) ([]ADBUser, error) {
+	return getUsers(db, options)
+}
+
+func getUsers(db *sqlx.DB, options GetUserOptions) ([]ADBUser, error) {
+	query := selectUserBaseQuery
+
+	var queryArgs []interface{}
+
+	if options.ID != 0 {
+		query += " WHERE id = ? "
+		queryArgs = append(queryArgs, options.ID)
+	}
+
+	query += " ORDER BY email "
+
+	var users []ADBUser
+	if err := db.Select(&users, query, queryArgs...); err != nil {
+		return nil, errors.Wrapf(err, "failed to get users")
+	}
+
+	return users, nil
+}
+
+func buildUserJSONArray(users []ADBUser) []UserJSON {
+	var usersJSON []UserJSON
+
+	for _, u := range users {
+
+		usersJSON = append(usersJSON, UserJSON{
+			ID:       u.ID,
+			Email:    u.Email,
+			Admin:    u.Admin,
+			Disabled: u.Disabled,
+		})
+	}
+
+	return usersJSON
+}
+
+func CleanUserData(body io.Reader) (ADBUser, error) {
+	var userJSON UserJSON
+	err := json.NewDecoder(body).Decode(&userJSON)
+
+	if err != nil {
+		return ADBUser{}, err
+	}
+
+	if err := checkForDangerousChars(userJSON.Email); err != nil {
```

Comment:
We probably don't need this check. We only need it for data that ends up on non-vue pages, like new event and events list, because those pages don't properly escape values from the server.

Reply:
I didn't know that about Vue escaping input values. Good to know :) 

I was surprised to see the need for this, as I thought the router would have something built in. I'll go ahead and remove it. Especially since only "admins" will be using the endpoint that calls this method.